viewing_party/party.py

```diff
@@ -1,23 +1,168 @@
+    if user_data:
+        for i in user_data["watched"]:
+            genre_list.append(i['genre'])
+        for i in genre_list:
+            if i not in genre_tally_dict:
+                genre_tally_dict[i] = 1
+                most_watched_int = 1
+            else:
+                genre_tally_dict[i] += 1
+                most_watched_int += 1
+            for i in genre_tally_dict:
+                if genre_tally_dict[i] > most_watched_int:
+                    most_watched_str = i
+        return most_watched_str
+    return None
+
+    # -----------------------------------------
+    # ------------- WAVE 3 --------------------
+    # -----------------------------------------
+
+
+def get_unique_watched(user_data):
+    user_watched_list = user_data['watched']
+    friends = user_data['friends']
+    user_unique_watched_list = []
+    friends_title_list = []
+    for friend in friends:
+        for movie in friend['watched']:
+            friends_title_list.append(movie['title'])
+    for movie in user_watched_list:
+        if movie['title'] not in friends_title_list:
+            user_unique_watched_list.append(movie)
+    return user_unique_watched_list
+
+
+def get_friends_unique_watched(user_data):
+
+    user_watched_list = user_data['watched']
+    friends = user_data['friends']
+    friends_unique_movie_list = []
+    user_title_list = []
+    for movie in user_watched_list:
+        user_title_list.append(movie['title'])
+    for friend in friends:
+        for movie in friend['watched']:
+            if movie['title'] not in user_title_list and movie not in friends_unique_movie_list:
+                friends_unique_movie_list.append(movie)
+    return friends_unique_movie_list
+
+    # -----------------------------------------
+    # ------------- WAVE 4 --------------------
+    # -----------------------------------------
+
+
+def get_available_recs(user_data):
+
+    subscriptions_list = user_data['subscriptions']
+    user_unique_watched_list = get_unique_watched(user_data)
+    friends_unique_movie_list = get_friends_unique_watched(
+        user_data)  # helper func
+    list_of_recommended_movies = []
+
+    for movie in friends_unique_movie_list:
+        if movie not in user_unique_watched_list and movie['host'] in subscriptions_list:
+            list_of_recommended_movies.append(movie)
+
+    return list_of_recommended_movies
+
+    # -----------------------------------------
+    # ------------- WAVE 5 --------------------
+    # -----------------------------------------
+
+
+def get_new_rec_by_genre(user_data):
+
+    friends_unique_watched_list = get_friends_unique_watched(user_data)
+    users_favorite_genres = get_most_watched_genre(user_data)
+    list_of_recommended_genre_movies = []
+    if users_favorite_genres and friends_unique_watched_list:
+        for movie in friends_unique_watched_list:
+            if movie['genre'] in users_favorite_genres and movie in friends_unique_watched_list:

```

Comment:
All we need here is
```py
            if movie['genre'] == users_favorite_genre:  # note the renamed variable
```
Sionce the genre is expected to be a string, we should do value comparison with `==`. `in` appears to work, since with a string, python does a substring check.

For example, `"abc" in "abcd"` is `True`. And any string is a substring of itself, so `"abc" in "abc"` is also `True`. But we just want to check the exact value.

We already know that `movie` is in `friends_unique_watched_list` since that's the list `movie` is iterating over.